emc-metalnx-services/src/test/java/com/emc/metalnx/services/irods/CollectionServiceImplTest.java

```diff
@@ -140,5 +141,46 @@ public void testGetSubCollectionsAndDataObjectsUnderPathThatMatchSearchTextPagin
 		Assert.assertTrue("no recs returned", actual.size() > 1);
 
 	}
+	
+	@Test
+	public void testCreateCollectionWithParent() throws Exception {
+		
+		IRODSAccount irodsAccount = testingPropertiesHelper.buildIRODSAccountFromTestProperties(testingProperties);
+
+		IRODSServicesImpl irodsServices = new IRODSServicesImpl();
+		irodsServices.setIrodsAccount(irodsAccount);
+		irodsServices.irodsAccessObjectFactory = irodsFileSystem.getIRODSAccessObjectFactory();
+
+		CollectionServiceImpl collectionService = new CollectionServiceImpl();
+		collectionService.setIrodsServices(irodsServices);
+
+		// get new collection as DataGridCollectionAndDataObject
+		String newSubCollection = "subdir1/subdir2";
+		StringBuilder parentCollectionStringBuilder = new StringBuilder();
+		parentCollectionStringBuilder.append(irodsAccount.getHomeDirectory());
+		//parentCollectionStringBuilder.append("/subdir1");
+		
+		String currentPath = irodsAccount.getHomeDirectory();
+		String fullPath = currentPath + '/' + newSubCollection;
+		
+		DataGridCollectionAndDataObject newCollection = new DataGridCollectionAndDataObject(
+				fullPath, newSubCollection, currentPath, true);
+
+		newCollection.setParentPath(currentPath);
+		newCollection.setCreatedAt(new Date());
+		newCollection.setModifiedAt(newCollection.getCreatedAt());
+		newCollection.setInheritanceOption(false);
+
+		// create new collection
+		collectionService.createCollection(newCollection, true);
+		
+		// search for new collection
+		StringBuilder sb = new StringBuilder();
+		sb.append(irodsAccount.getHomeDirectory());
+		sb.append("newSubCollection");
```

Comment:
Looks like these lines can be removed too.

Reply:
done